### PR TITLE
Refactor/content creation routes

### PR DESCRIPTION
**Pull request by agusbusta**
Enhanced Analysis API Endpoints with Dynamic Table Selection

This PR introduces significant improvements to the Analysis API endpoints, implementing a dynamic table selection mechanism based on the section target. The changes affect the GET, POST, DELETE, and PUT routes for analyses.

Key Changes:
1. Added 'section_id' as a required parameter for all affected routes.
2. Implemented logic to determine the correct database table using the section's target.
3. Modified queries to use the dynamically selected model class.
4. Updated error handling and response messages to reflect the specific table operations.

Affected Routes:
- GET /analysis: Now retrieves analyses from the correct table based on section_id and coin_bot_id.
- POST /analysis: Creates a new analysis in the appropriate table determined by section_id.
- DELETE /analysis/<int:analysis_id>: Deletes an analysis from the appropriate table determined by section_id.
- PUT /analysis/<int:analysis_id>: Updates an analysis in the correct table as specified by section_id.

### CODE REVIEW

**Comment by daviddflix** [comment]
### Feedback 

Hey @agusbusta, The PR has a conflict. Also, migration files are just not right, could you please check them as there are 10 of them and some of them do nothing or remove the apscheduler column, and drop column, I believe this branch which is for upgrading the analysis endpoint does not need any migration or any column to be updated regarding the analysis table or am I wrong. either way could you please curate the migration files if required?

File: alembic/versions/574453f8f426_sandr_table_created.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# revision identifiers, used by Alembic.
+revision: str = '574453f8f426'
+down_revision: Union[str, None] = '5d2cde0197b2'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index('ix_apscheduler_jobs_next_run_time', table_name='apscheduler_jobs')
```
Comment (daviddflix) [inline alembic/versions/574453f8f426_sandr_table_created.py:23]: This migration file is trying to delete something that does not exist

```
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1848, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1988, in _exec_single_context
    self._handle_dbapi_exception(
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 2343, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1969, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\default.py", line 922, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedObject) no existe el índice «ix_apscheduler_jobs_next_run_time»

[SQL:
DROP INDEX ix_apscheduler_jobs_next_run_time]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

File: alembic/versions/095f9e28f2f5_sandr_table_created.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# revision identifiers, used by Alembic.
+revision: str = '095f9e28f2f5'
+down_revision: Union[str, None] = '5ff73cd63101'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index('ix_apscheduler_jobs_next_run_time', table_name='apscheduler_jobs')
```
Comment (daviddflix) [inline alembic/versions/095f9e28f2f5_sandr_table_created.py:23]: This migration file is also trying to remove something that does not exist

```
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1988, in _exec_single_context
    self._handle_dbapi_exception(
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 2343, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1969, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Users\David\AppData\Local\Programs\Python\Python312\Lib\site-packages\sqlalchemy\engine\default.py", line 922, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedObject) no existe el índice «ix_apscheduler_jobs_next_run_time»

[SQL:
DROP INDEX ix_apscheduler_jobs_next_run_time]
(Background on this error at: https://sqlalche.me/e/20/f405)
```

**Review by daviddflix — CHANGES_REQUESTED** [review]
**Important**

- Create a new file sections.py inside the folder routes/analysis/sections.py.
- Create three new endpoints:

**GET: To obtain all sections.
POST: to create a new section
DELETE: delete a secion**


**Update the documentation to reflect these changes.**

Why these endpoints: We are requesting the section_id in the analysis endpoints.

**Comment by agusbusta** [comment]
Fixed and routes created and tested!

**Comment by daviddflix** [comment]
Feedback

- Delete unnecessary migration files
- Create the right migration script based on the schemas added in this branch.
- Review documentation.

**Comment by agusbusta** [comment]
changes are done!
Migration files that looks like unnecessary are being implicitly used by others in dependency, that's why I can't delete them but I merged and fixed all

**Review by daviddflix — APPROVED** [review]

**Pull request merged**